js/customize-snapshots-front.js

```diff
@@ -0,0 +1,72 @@
+/* global jQuery, wp */
+/* exported CustomizeSnapshotsFront */
+var CustomizeSnapshotsFront = (function( $ ) {
+	'use strict';
+
+	var component = {
+		data: {
+			confirmationMsg: '',
+			action: '',
+			snapshotsFrontendPublishNonce: '',
+			errorMsg: ''
+		}
+	};
+	/**
+	 * Init.
+	 *
+	 * @param {object} args Args.
+	 * @returns {void}
+	 */
+	component.init = function init( args ) {
+		_.extend( component.data, args );
+
+		$( document ).ready( function() {
+			component.setupPublishButton();
+		} );
+	};
+
+	/**
+	 * Set up snapshot frontend publish button.
+	 *
+	 * @returns {void}
+	 */
+	component.setupPublishButton = function setupPublishButton() {
+		var publishBtn = $( '#wp-admin-bar-publish-customize-snapshot a' );
+
+		if ( ! publishBtn.length ) {
+			return;
+		}
+
+		publishBtn.click( function( e ) {
+			var request,
+				data = {
+					nonce: component.data.snapshotsFrontendPublishNonce,
+					uuid: component.data.uuid
+				};
+			e.preventDefault();
+
+			if ( ! window.confirm( component.data.confirmationMsg ) ) { // eslint-disable-line no-alert
+				return false;
+			}
+			if ( ! wp.customize.settings.theme.active ) {
```

Comment:
Seems like the issue was because I was using the param `changeset_uuid=...` instead of `customize_changeset_uuid` in frontend.

Note that `customize-snapshots-frontend.js` file was added to `develop` branch so I merged the frontend JS file from this branch (`customize-snapshots-front.js`) into the one in `develop` (`customize-snapshots-frontend.js`). Let me know if there are any concerns with that. Also added removing the UUID from storage after publishing.

There is another issue with publishing the changesets from frontend:
`WP_Customize_Setting::_preview_filter` is returning the dirty value as old value instead of the actual old value in DB meaning that the setting is not saved when publishing.

Although I can see that `Post_Type::hooks()` is adding this filter:
`add_action( 'transition_post_status', array( $this, 'start_pretending_customize_save_ajax_action' ), $priority - 1, 3 );`, this method is always returning true: `WP_Customize_Setting::is_current_blog_previewed()`.

Looks like `WP_Customize_Manager` is initialized in `_wp_customize_include()` before the filters have any effect. Also, not sure if https://github.com/xwp/wordpress-develop/pull/231 would fix that since the `$_REQUEST['action']` is not `customize_save` at that moment.

Am I missing something or is creating another workaround necessary, e.g. reinitializing `WP_Customize_Manager`?